aeneas/src/ssa/SsaBlockOrder.v3

```diff
@@ -30,25 +35,40 @@ class SsaBlockOrder {
 		}
 		this.list = null;
 		this.loopEdges = null;
+
 	}
-	// post-number a block recursively
+	// post-number a block recursively (also build a list of the post ordering of the nodes)
 	private def number(s: SsaBlock) {
-		s.mark = ON_STACK;
+		// marker.setMark(s, ON_STACK);
+		if (ON_STACK == 0) {
```

Comment:
I think you can allocate the values for ON_STACK and DONE in the constructor of this class.